issue #47

asd24684782 opened the pull request:
## 描述
#47 
- [ ] User email verify
- [ ] Password reset
## 變更類型

- [ ] 新功能（增加功能的非破壞性更改

## 這是如何測試的?

- [ ] Click verify url and set user.is_verifyed `True`.
- [ ] Verify api with wrong uid or token will return 400.
- [ ] Verify api will return 400 if user is verifyed.
- [ ] Reset password api send email if user exist.
- [ ] Check reset password success, compare new password with database.

Commit:
send verify email on register and verify api

Commit:
fetch new version

Commit:
shell

Commit:
style(setting.py,-ContestDetailView): lint

Commit:
fix(user/views.py): change http method

Commit:
fix(user/views.py): lint and add check user is exist or is verified

Commit:
test(test_user_api.py): test email verify

Commit:
Merge branch 'develop' of https://github.com/asd24684782/django-backend into develop

Commit:
feat(user/url.py-seriallizers.py-template,-app/setting.py,-requirements): send email reset password method

almost use django_rest_passwordreset package, only write email content by myself in serializers.py.

Commit:
test(user/test): add reset password test

check reset password user exist and check reset password success

jianan1104 commented:
hey, @asd24684782 Lawrence. Thanks for your help. I really appreciate you. Just test and looks good, but here are a few things I would like you to do. It'll have a better development experience for the next developer to interact with these functions. Check it below, please.

- Add 2 variables to `.env.sample`
  - Set the E-mail Account name to "Fingerstyle Taiwan"
  - `EMAIL_HOST_USER=example@gmail.com`
  - `EMAIL_HOST_PASSWORD=GmailAppPassword`
- Configuration for different environments 
  - When `DEBUG=true` *(dev)*
      - keep the same settings
  -  When `DEBUG=false` *(prod)*
      - `EMAIL_PORT = 465`  *because backend service is running on SSL.*
      - Change `{{domain}}` to `www.fingerstyletaiwan.com` at `verify_email.html`. *because we would use a frontend page to do it.*
      - Change the URL for `reset_password_url.html` *same reason as above*

If you have a question, please feel free to leave a message.

Commit:
fix(.env.sample,-setting.py-user/serializers.py,-user/template/*.html): add email info in env.sample and add env for prod

combine email content url at backend in serializers.py make email template more clear.
